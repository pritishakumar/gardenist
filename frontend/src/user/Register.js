import React from 'react';
import { registerActionThunk } from '../redux/actionCreators';
import { useDispatch } from 'react-redux';
// Components from within this project
import UserForm from './UserForm';
// Material UI Components
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

/** Component that is rendered by ./Home Component.
 * It is a wrapper Component, that renders and 
 * customizes ./UserForm Component to register the 
 * user
 * 
 * Authorization: none, accessible for all
 */
function Register() {
	const dispatch = useDispatch();

	async function registerHandler(data) {
		const result = await dispatch(registerActionThunk(data));
		if (result) {
			return 'Registration successful!';
		}
	}

	return (
		<Card className="Register">
			<CardContent align="center">
				<Typography variant="h3">Register for an Account!</Typography>
				<UserForm actionHandler={registerHandler} mode="Register" />
			</CardContent>
		</Card>
	);
}

export default Register;
