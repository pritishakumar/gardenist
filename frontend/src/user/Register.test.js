import React from "react";
import { MemoryRouter } from 'react-router-dom';
import { render, fireEvent } from "@testing-library/react";
import Register from "./Register";
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

// smoke test
it("renders without crashing", function() {
  const mockStore = configureStore();
  const initialState = {};
  const store = mockStore(initialState);

  render(
    <Provider store={store}>
      <MemoryRouter initialEntries = {['/']}>
        <Register />
      </MemoryRouter>
    </Provider>
  );
});

// snapshot test
it("matches snapshot", function() {
  const mockStore = configureStore();
  const initialState = {};
  const store = mockStore(initialState);

  const { asFragment } =  render(
    <Provider store={store}>
      <MemoryRouter initialEntries = {['/']}>
        <Register />
      </MemoryRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
})