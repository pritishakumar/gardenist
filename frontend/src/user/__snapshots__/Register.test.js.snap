// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root Register MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      align="center"
      class="MuiCardContent-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        Register for an Account!
      </h3>
      <div
        class="UserForm"
      >
        <form
          action="#"
        >
          <div
            class="MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
              data-shrink="false"
              for="email"
            >
              Email Address: 
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input-2 MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                name="email"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
              data-shrink="false"
              for="name"
            >
              First Name: 
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input-2 MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                name="name"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
              data-shrink="false"
              for="password"
            >
              Password: 
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input-2 MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                name="password"
                type="password"
                value=""
              />
            </div>
          </div>
          <br />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-input-2 MuiButton-containedPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Register
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
